packages/node_modules/@webex/plugin-meetings/src/meetings/index.js

```diff
@@ -57,8 +87,23 @@ export default class Meetings extends SparkPlugin {
       const {eventType} = data;
       console.log('Mercury-Locus-Event', eventType);
       if (eventType !== LOCUSEVENT.MESSAGE_ROAP) {
-        const meeting = this.getMeetingByType(LOCUS_URL, data.locusUrl);
-        this.parser.parse(meeting, data);
+        this.handleLocusEvent(data);
+      }
+    });
+
+
+    this.spark.internal.mercury.on(ROAP.ROAP_MERCURY, (envelope) => {
+      const {data} = envelope;
+      const {eventType} = data;
+      if (eventType === LOCUSEVENT.MESSAGE_ROAP) {
+        const meeting = this.getMeetingByType(CORRELATION_ID, data.correlationId);
+        if (meeting) {
+          meeting.roap.roapEvent(data);
```

Comment:
Wasn't this handled with RoapHandler before? this method does the same thing?

Reply:
Arun please comment

Reply:
ya moved here because the event was being register  multiple times . In this place it gets registered only once

Reply:
cool